src/Versity.Users/External/Presentation/Controllers/AuthController.cs

```diff
@@ -0,0 +1,48 @@
+﻿using System.Security.Claims;
+using Application.Dtos;
+using Application.RequestHandlers.Auth.Commands.GetAdminRole;
+using Application.RequestHandlers.Auth.Commands.LoginVersityUser;
+using Application.RequestHandlers.Auth.Commands.RegisterVersityUser;
+using MediatR;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using Presentation.Abstractions;
+
+namespace Presentation.Controllers;
+
+[Route("api/[controller]/[action]")]
+public sealed class AuthController : ApiController
+{
+    public AuthController(ISender sender) : base(sender)
+    {
+    }
+
+    [HttpPost]
+    public async Task<IActionResult> Register(RegisterVersityUserDto userDto, CancellationToken cancellationToken)
+    {
+        var command = new RegisterVersityUserCommand(userDto.FirstName, userDto.LastName, userDto.Email, userDto.Phone, userDto.Password);
+        var result = await Sender.Send(command, cancellationToken);
+        return result.Succeeded ? Ok() : BadRequest(result.Errors);
+    }
+    
+    [HttpPost]
+    public async Task<IActionResult> Login(LoginVersityUserDto userDto, CancellationToken cancellationToken)
+    {
+        var command = new LoginVersityUserCommand(userDto.Email, userDto.Password);
+        var token = await Sender.Send(command, cancellationToken);
+        return Ok(new { Token = token });
+    }
+    
+    [Authorize]
+    [HttpPost]
+    public async Task<IActionResult> GetAdminRole(CancellationToken cancellationToken)
+    {
+        var userId = HttpContext.User.Claims.First(x => x.Type == ClaimTypes.NameIdentifier).Value;
```

Comment:
Move it to business logic

Reply:
Claims are now checked [in the command handler](https://github.com/jejikeh/versity/pull/12/commits/7e2160918bb47ecfc91a43dff0003b8f6c931d9b#diff-c87bc9d31cf6ee833c66d1962475c7bbf6637811430cb40cfdc9c17448367dac).

Maybe it would be better to put the claims verification logic in a separate service altogether and inject it instead of IHttpContextAccessor?